sdk/tests/wallet/chrysalis_migration.rs

```diff
@@ -0,0 +1,221 @@
+// Copyright 2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{fs, io, path::Path};
+
+use iota_sdk::{
+    client::{constants::IOTA_COIN_TYPE, secret::SecretManager, Password},
+    types::block::address::ToBech32Ext,
+    wallet::{migration::migrate_db_chrysalis_to_stardust, ClientOptions, Result},
+    Wallet,
+};
+
+use crate::wallet::common::{setup, tear_down};
+
+#[tokio::test]
+async fn migrate_chrysalis_db() -> Result<()> {
+    let storage_path = "migrate_chrysalis_db/db";
+    setup(storage_path)?;
+    // Copy db so the original doesn't get modified
+    copy_folder("./tests/wallet/fixtures/chrysalis-db/db", storage_path).unwrap();
+
+    migrate_db_chrysalis_to_stardust("migrate_chrysalis_db".into(), None).await?;
+
+    let client_options = ClientOptions::new();
+    let wallet = Wallet::builder()
+        .with_storage_path("migrate_chrysalis_db")
+        .with_client_options(client_options)
+        .finish()
+        .await?;
+
+    let accounts = wallet.get_accounts().await?;
+    assert_eq!(accounts.len(), 2);
+    assert_eq!(accounts[0].alias().await, "Alice");
+    assert_eq!(accounts[1].alias().await, "Bob");
+
+    let alice_acc_details = accounts[0].details().await;
+    assert_eq!(alice_acc_details.public_addresses().len(), 2);
+    assert_eq!(
+        alice_acc_details.public_addresses()[0].address().try_to_bech32("rms")?,
+        "rms1qqqu7qry22f6v7d2d9aesny9vjtf56unpevkfzfudddlcq5ja9clv44sef6"
+    );
+    assert_eq!(alice_acc_details.internal_addresses().len(), 1);
+    assert_eq!(
+        alice_acc_details.internal_addresses()[0]
+            .address()
+            .try_to_bech32("rms")?,
+        "rms1qz4tac74vympq4hqqz8g9egrkhscn9743svd9xxh2w99qf5cd8vcxrmspmw"
+    );
+
+    let bob_acc_details = accounts[1].details().await;
+    assert_eq!(bob_acc_details.public_addresses().len(), 1);
+    assert_eq!(
+        bob_acc_details.public_addresses()[0].address().try_to_bech32("rms")?,
+        "rms1qql3h5vxh2sxa93yadh7f4rkr7f9g9e65wlytazeu688mpcvhvmd2xvfq8y"
+    );
+    assert_eq!(bob_acc_details.internal_addresses().len(), 1);
+    assert_eq!(
+        bob_acc_details.internal_addresses()[0].address().try_to_bech32("rms")?,
+        "rms1qq4c9kl7vz0yssjw02w7jda56lec4ss3anfq03gwzdxzl92hcfjz7daxdfg"
+    );
+
+    let chrysalis_data = wallet.get_chrysalis_data().await?;
+    assert!(!chrysalis_data.unwrap().is_empty());
```

Comment:
We should be able to test at least one thing from this?

Reply:
Added one